src/library.js

```diff
@@ -2078,16 +2078,11 @@ addToLibrary({
   // at runtime.
   $keepRuntimeAlive__deps: ['$runtimeKeepaliveCounter'],
   $keepRuntimeAlive: () => noExitRuntime || runtimeKeepaliveCounter > 0,
-#elif !EXIT_RUNTIME
-  // When `noExitRuntime` is not include and EXIT_RUNTIME=0 then we know the
+#elif !EXIT_RUNTIME && !PTHREADS
+  // When `noExitRuntime` is not included and EXIT_RUNTIME=0 then we know the
   // runtime can never exit (i.e. should always be kept alive).
-  // However for pthreads we always default to allowing the runtime to exit
-  // otherwise threads never exit and are not joinable.
-#if PTHREADS
-  $keepRuntimeAlive: () => !ENVIRONMENT_IS_PTHREAD,
-#else
+  // However, for pthreads we always always need to allow the runtime to exit
```

Comment:
Why do we need to always allow the runtime to exit with pthreads? Is that documented somewhere?

And, if so, should we do pthreads => `EXIT_RUNTIME=1`?

Reply:
Because pthreads themselves need to be able to exit.    `EXIT_RUNTIME=0` means the main thread never exits but it doesn't mean that pthreads can't exit.

Each pthread has its own "runtime" that can be kept alive and stops the thread from exiting.   `EXIT_RUNTIME` has not effect (or should have no effect) on threads.

Reply:
I updated the comment.